src/Plugin/Validation/Constraint/UniqueOgMembershipConstraintValidator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\og\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Ensures that new members added to a group do not already exist.
+ *
+ * Note that in typical operation, this validation constraint will not come into
+ * play, as the membership entity's uid field is already validated by core's
+ * ValidReferenceConstraint, which hands over to the entity reference selection
+ * plugin. In our case, that is
+ * \Drupal\og\Plugin\EntityReferenceSelection\OgUserSelection, which already
+ * checks an existing member cannot be added to the group again.
+ */
+class UniqueOgMembershipConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    /* @var \Drupal\Core\Field\FieldItemInterface $value */
+    if (!isset($value)) {
+      return;
+    }
+
+    /* @var \Drupal\og\Entity\OgMembership $entity */
+    $entity = $value->getEntity();
+
+    // Only applicable to new memberships.
+    if (!$entity->isNew()) {
+      return;
+    }
+
+    // The default entity reference constraint adds a violation in this case.
+    $value = $value->getValue();
+    if (!isset($value[0]) || !isset($value[0]['target_id'])) {
+      return;
+    }
+
+    $new_member_uid = $value[0]['target_id'];
+
+    $query = \Drupal::service('entity_type.manager')
+      ->getStorage('og_membership')
+      ->getQuery()
+      ->condition('uid', $new_member_uid);
+    $membership_ids = $query->execute();
+
+    if ($membership_ids) {
```

Comment:
If there are memberships, the constraint is thrown. This causes a bug that a user can only be in one group.

Reply:
Added you as a collaborator to my OG fork. I'm hoping to get some time to work on this soon, but int he mean time can you commit your fix?

Reply:
Thank you, and pushed!